src/mpid/ch4/netmod/ofi/ofi_recv.h

```diff
@@ -104,12 +104,13 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_irecv(void *buf,
     MPIR_Request *rreq = *request;
     uint64_t match_bits, mask_bits;
     MPIR_Context_id_t context_id = comm->recvcontext_id + context_offset;
-    size_t data_sz;
+    size_t data_sz = 0;
```

Comment:
Why do we initialize to 0 here? If there is a warning on uninitialized usage, we need set values in each code path. Otherwise, how do we know that this `0` works for each code path?

Reply:
In case you fail at line #124, data_sz is still uninitialized over there, but in `fn_fail`, data_sz is used for condition check.

Reply:
Also, `data_sz` is overwritten at line 133. Compiler just complains that it is possible for `fn_fail` check before line 133.

Reply:
I see. How about use a flag so we don't have to repeat the condition check in `fn_fail`? When you repeat the condition check, whenever we update the condition logic, we need modify it twice, that's an unnecessary maintenance overhead.

Reply:
So you wanna a flag showing there is allocated CPU staging buffer, right?

Reply:
Yes. That way it is explicit.

Reply:
I can add that, same for `ofi_send.h`.